crates/dock/tests/mutability_check.rs

```diff
@@ -0,0 +1,88 @@
+#[cfg(test)]
+mod tests {
+    use expect_test::expect_file;
+    use regex::Regex;
+
+    fn check(dir_name: &str) {
+        let mut out = Vec::new();
+        let mut err = Vec::new();
+
+        let main_file_contents =
+            std::fs::read_to_string(format!("tests/mutability_check/{dir_name}/main.nail"))
+                .unwrap();
+        // //---stdoutで区切ることで、標準出力と標準エラー出力をテストする
+        // //---stderrは必ず//---stdoutより後に書くことが前提
+        let (main_contents, _) = main_file_contents
+            .split_once("//---stdout")
+            .unwrap_or((&main_file_contents, ""));
+        let mut main_contents = main_contents.trim().to_string();
+
+        match dock::execute(
+            dock::NailExecutablePath::RootFile(
+                std::env::current_dir()
+                    .unwrap()
+                    .join(format!("tests/mutability_check/{dir_name}/main.nail")),
+            ),
+            &mut out,
+            &mut err,
+            false,
+        ) {
+            Ok(_) => (),
+            Err(e) => match e {
+                // 型チェックのエラーはExecutionError::Nailで返るため何もしない
+                // それ以外は型チェック以外のエラーなのでpanicしてテストを失敗させる
+                dock::ExecutionError::Nail => (),
+                dock::ExecutionError::Io(e) => panic!("io error: {e}"),
+                dock::ExecutionError::InvalidRootNailFilePath(e) => {
+                    panic!("invalid root nail file path: {e}")
+                }
+            },
+        }
```

Comment:
**Performance Issue**: The execution function is called without proper error handling for different types of ExecutionError, which can lead to unhandled exceptions and affect the robustness and reliability of the test execution flow. <br> **Fix**: Implement comprehensive error handling for each specific type of ExecutionError to ensure that all potential issues are gracefully handled and logged appropriately. <br> **Code Suggestion**: 
 ```
 +            Err(e) => match e {
 +                dock::ExecutionError::Nail => log::error!("Nail execution error occurred"),
 +                dock::ExecutionError::Io(e) => {
 +                    log::error!("IO error during execution: {}", e);
 +                    return Err(e);
 +                },
 +                dock::ExecutionError::InvalidRootNailFilePath(e) => {
 +                    log::error!("Invalid file path: {}", e);
 +                    return Err(e);
 +                },
 +            }
 ```